processor/src/lib.rs

```diff
@@ -152,27 +152,28 @@ where
     // CONSTRUCTORS
     // --------------------------------------------------------------------------------------------
     /// Creates a new process with the provided inputs.
-    pub fn new(kernel: &Kernel, stack_inputs: StackInputs, advice_provider: A) -> Self {
+    pub fn new(kernel: Kernel, stack_inputs: StackInputs, advice_provider: A) -> Self {
         Self::initialize(kernel, stack_inputs, advice_provider, false)
     }
 
     /// Creates a new process with provided inputs and debug options enabled.
-    pub fn new_debug(kernel: &Kernel, stack_inputs: StackInputs, advice_provider: A) -> Self {
+    pub fn new_debug(kernel: Kernel, stack_inputs: StackInputs, advice_provider: A) -> Self {
         Self::initialize(kernel, stack_inputs, advice_provider, true)
     }
 
     fn initialize(
-        kernel: &Kernel,
+        kernel: Kernel,
         stack: StackInputs,
         advice_provider: A,
         in_debug_mode: bool,
     ) -> Self {
+        let chiplets = Chiplets::new(kernel);
```

Comment:
Could we not keep it the the same way as it was before (i.e., constructed on line 176).

Reply:
True, the previous iteration required the kernel to be constructed before with the chiplet, so this was a leftover